lib/palindrome_check.rb

```diff
@@ -1,5 +1,24 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+  y = (my_phrase.length - 1)
+  x = 0
+  palindrome = true
+  until x > y do
+    until my_phrase[x] =~ /\S/ do
+      x += 1
+    end
+    until my_phrase[y] =~ /\S/ do
```

Comment:
As further due diligence and to make your algorithm robust, ensure that x is less than or equal to y before comparing the character at y with a white space in this loop.